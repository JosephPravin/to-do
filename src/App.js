import React, { Component } from "react";
import './App.css';
import Users from './components/users';
import AddUser from './components/addUser';
import NavBar from './components/navBar';
import users from './data/users.json'

class App extends Component {
  state = {
    users: Array
  };

  constructor(props) {
    super(props);
    this.state = { users };
  };

  handleAddUser = user => {
    const users = [...this.state.users];
    users.push({id:users.length+1,name: user, tasks: []})
    this.setState({ users });
  }

  handleAddTask = (user, task) => {
    const users = [...this.state.users];
    const index = users.findIndex(u=>u.name===user);
    users[index].tasks.push(task);  
    this.setState({users: users})
  }

  handleDeleteTask = (user, task) => {
    const users = [...this.state.users];
    const userIndex = users.findIndex(u=>u.name===user);
    const taskIndex = users[userIndex].tasks.indexOf(users[userIndex].tasks.filter(t => t.title === task.title)[0]);
    users[userIndex].tasks.splice(taskIndex, 1);
    this.setState({users})
  }

  render() {     
    return (
      
        <div>
        <NavBar header="To Do app" users={ this.state.users }></NavBar>
       
        <div className="row">
          <div className="col-10"><Users users={this.state.users} onAddTask={this.handleAddTask} onDeleteTask={this.handleDeleteTask}/></div>
          <div className="col-2" style={{paddingLeft: '0px', marginTop: '10px'}}><AddUser onAddUser={this.handleAddUser}/></div>
        </div>
        </div> 

    );
  }
}
 
export default App;
