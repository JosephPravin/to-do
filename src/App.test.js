import React from 'react';
import renderer from "react-test-renderer";
import App from './App';

test("App snapshot", () => {
  const component = renderer.create(<App />);
  expect(component.toJSON()).toMatchSnapshot();
});
