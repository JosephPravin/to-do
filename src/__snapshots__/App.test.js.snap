// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<div>
  <nav
    className="navbar navbar-dark bg-dark"
  >
    <a
      class="navbar-brand"
      href="#"
    >
      To Do app
    </a>
    <span
      class="btn btn-outline-light my-2 my-sm-0"
    >
      2
       users have tasks
    </span>
  </nav>
  <div
    className="row"
  >
    <div
      className="col-10"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col"
            style={
              Object {
                "padding": "1em",
              }
            }
          >
            <h3>
              <mark>
                Joseph
              </mark>
            </h3>
            <div>
              <div
                className="card"
                style={
                  Object {
                    "width": "18rem",
                  }
                }
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Button link inactive
                  </h5>
                  <p
                    className="card-text bg-light"
                  >
                    Though home page button should be active, it is inactive on page load
                  </p>
                  <a
                    class="card-link"
                    href="#"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "red",
                        "float": "right",
                      }
                    }
                  >
                    Delete
                  </a>
                </div>
              </div>
            </div>
            <button
              className="btn btn-sm btn-dark"
              onClick={[Function]}
            >
              Add task
            </button>
          </div>
          <div
            className="col"
            style={
              Object {
                "padding": "1em",
              }
            }
          >
            <h3>
              <mark>
                Pravin
              </mark>
            </h3>
            <div>
              <div
                className="card"
                style={
                  Object {
                    "width": "18rem",
                  }
                }
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Jenkins task
                  </h5>
                  <p
                    className="card-text bg-light"
                  >
                    Create a jenkins job for UI deployment in dev. Branch name should be parameterised
                  </p>
                  <a
                    class="card-link"
                    href="#"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "red",
                        "float": "right",
                      }
                    }
                  >
                    Delete
                  </a>
                </div>
              </div>
              <div
                className="card"
                style={
                  Object {
                    "width": "18rem",
                  }
                }
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Text overflowing in title card
                  </h5>
                  <p
                    className="card-text bg-light"
                  >
                    Css changes to accomodate the text within the div space
                  </p>
                  <a
                    class="card-link"
                    href="#"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "red",
                        "float": "right",
                      }
                    }
                  >
                    Delete
                  </a>
                </div>
              </div>
            </div>
            <button
              className="btn btn-sm btn-dark"
              onClick={[Function]}
            >
              Add task
            </button>
          </div>
          <div
            className="col"
            style={
              Object {
                "padding": "1em",
              }
            }
          >
            <h3>
              <mark>
                Rajaram
              </mark>
            </h3>
            <div />
            <button
              className="btn btn-sm btn-dark"
              onClick={[Function]}
            >
              Add task
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-2"
      style={
        Object {
          "marginTop": "10px",
          "paddingLeft": "0px",
        }
      }
    >
      <div>
        <button
          className="btn btn-primary"
          onClick={[Function]}
        >
          + Add new user
        </button>
      </div>
    </div>
  </div>
</div>
`;
